src/neo/Network/P2P/ProtocolHandler.cs

```diff
@@ -135,7 +135,14 @@ private void OnMessage(Message msg)
                     break;
                 case MessageCommand.Transaction:
                     if (msg.Payload.Size <= Transaction.MaxTransactionSize)
-                        OnInventoryReceived((Transaction)msg.Payload);
+                    {
+                        Transaction tx = (Transaction)msg.Payload;
+                        if (tx.Verify(Blockchain.Singleton.currentSnapshot, Blockchain.Singleton.MemPool.SendersFeeMonitor.GetSenderFee(tx.Sender)) != RelayResultReason.Succeed)
+                        {
```

Comment:
It's not safe here.

Reply:
Have seperate sender fee check from transaction verication and thus avoid unsafe situation, plz have a check:)

Reply:
Maybe we need a better name for `Verify`, as we discussed, verify split for two party: stateful and stateless.
Is it possible, move transaction.verify out of blockchain.cs, just RemoteNode/ProtocolHandler --- Enter ---> Mempool,  then Blockchain.cs will be simpler.

Reply:
Have update correspondingly, please have a look